### PR TITLE
sciter value improvements

### PR DESCRIPTION
**Pull request by pravic**

### CODE REVIEW

**Comment by SonnyX** [comment]
LGTM, is that todo high priority or not?

**Review by SonnyX — APPROVED** [review]

File: src/value.rs [hunk 1]
```diff
@@ -998,6 +1023,24 @@ impl<'a> From<&'a [u8]> for Value {
...
+impl From<std::time::SystemTime> for Value {
+	/// See the [VALUE_TIME::T_DATE].
+	fn from(val: std::time::SystemTime) -> Self {
+		let mut me = Value::new();
+		if let Ok(epoch) = val.duration_since(std::time::UNIX_EPOCH) {
+			// in 100ns from epoch
+			let ns = epoch.as_nanos() / 100;
+
+			// does not fail because does not check the input
+			// TODO: is SystemTime UTC or not?
```
Comment (pravic) [inline src/value.rs:1036]: It is, according to [`SystemTime`: underlaying system calls](https://doc.rust-lang.org/nightly/std/time/struct.SystemTime.html#underlying-system-calls) and [`GetSystemTime`](https://docs.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getsystemtimeasfiletime).

**Comment by pravic** [comment]
@SonnyX this PR is in progress, I wanted to add some more `const fn` and other `Value` improvements.

**Comment by SonnyX** [comment]
Just putting a second pair of eyes across any commits :)

File: src/value.rs (hunk 1 above)
Comment (pravic) [reply src/value.rs:1035]: But in Rust itself it's unspecified - see the other platforms' docs.

**Pull request merged**